app/controllers/api/v1/documents.rb

```diff
@@ -1,5 +1,18 @@
+# frozen_string_literal: true
+
+require 'mini_mime'
+
 module Api
   module V1
+    class ValidMimeType < Grape::Validations::Base
+      def validate_param!(attr_name, params)
+        return if MiniMime.lookup_by_content_type(params[attr_name])
+
+        raise Grape::Exceptions::Validation.new(params: [@scope.full_name(attr_name)],
+                                                message: 'is not a valid MIME type')
+      end
+    end
```

Comment:
@jamesmadhoc , this approach isn't ideal, because it duplicates the validation in the `Document` model. I suggest simplifying this by restricting the validation logic to the model. See https://github.com/MothOnMars/i14y/commit/749af63b714f080c691b17d7996f0911d0f7d497.

I14y is a little bit weird because the models and model-level validations were added very late in its life. This has led to what I now see are some inconsistencies and missing test cases for the controller. For now, please make the suggested changes, including the comment I've added re. the validations.

Reply:
@jamesmadhoc , I just realized I linked to an older version of my commit. Please add this comment to guide future devs:
https://github.com/MothOnMars/i14y/commit/5c2f1c6e0#diff-04c8891d20866a2eeb884bcce73deff08c6c923b3676d06bcd32dad6d4da57baR9-R11